stix2/sources/memory.py

```diff
@@ -283,7 +283,7 @@ def query(self, query=None, _composite_filters=None, allow_custom=False):
             if not isinstance(query, list):
                 # make sure dont make set from a Filter object,
                 # need to make a set from a list of Filter objects (even if just one Filter)
-                query = list(query)
+                query = [query]
```

Comment:
Can we replace this entire if block with just `query = list(query)`? I agree it's unnecessary with the isinstance check. In fact, we could combine with the below `query = set(list(query))` (which is a fairly common idiom I've seen). Some good tests would help ensure this part behaves as expected. In fact, we could probably pull out a function `def get_filter_set(query)` or put that as a method on the base `DataSource class`. 

We don't need to worry about this for this PR, but maybe make an issue for it?

Reply:
Correct me if I'm wrong, but the point seems to be to allow the method to accept either a single filter, or a list of filters.  A filter is itself an iterable (it's a tuple), so `list(query)`, if `query` is a single filter, would result in a list like `["prop", "=", "foo"]`, not what is intended.

Also, `set(list(query))` looks to me like it creates an unnecessary intermediate list.  Once you've ensured `query` is an iterable of filters, `set(query)` should create the filter set.

Reply:
@chisholm has the right of it. I got an error in the tests with `list(query)` but not `[query]`.

Reply:
Yes, you're right. Nothing to see here... move along 😁